aws-cpp-sdk-core/source/utils/threading/Executor.cpp

```diff
@@ -80,6 +84,7 @@ bool PooledThreadExecutor::SubmitToThread(std::function<void()>&& fn)
         m_tasks.push(fnCpy);
     }
 
+	std::unique_lock<std::mutex> locker(m_syncPointLock);

```

Comment:
I see the problem now. I disagree with the solution though.
I think it would be much better to encapsulate that synchronization logic in a Semaphore/AutoResetEvent class. I'll have something to review by tomorrow.
Thanks